### ISSUE TITLE
Enable Reverse Proxy Options

### ISSUE THREAD

**Issue by lmajano** [cfmlprojects/runwar#73]
@denuno What would it take to enable the reverse proxy options available in undertow: http://undertow.io/undertow-docs/undertow-docs-1.3.0/listeners.html

**Comment by denuno** [comment]
So what I was thinking was something along these lines:

You pass in just `--loadbalance "host:1,host:2:,host:3"` (host and port list of nodes) and it only load balances those-- they're already running somewhere.

If you also pass in the WAR path, then it fires up instances for each node, and balances those.  This is what it's doing now, and it gets the stop port from incrementing the node port.

I think each instance should be its own background process, versus a child of the load balancer process, otherwise they're sharing the same JVM instance I think, which has clustering implications.

We'll need to play around some more -- assuming this is even the ticket I think it is -- but I think we're pretty close to being able to fire off, with one command, a little cluster (heck, I'd like to support two load balancing instances, backed by 2 or 3 app node instances) that should be pretty fault tolerant, besides good for testing clustering stuff.

**Comment by lmajano** [comment]
Few comments here:

>You pass in just --loadbalance "host:1,host:2:,host:3" (host and port list of nodes) and it only load balances those-- they're already running somewhere.

That's perfect.  Is there a way to add/remove nodes from the balancing?

>I think each instance should be its own background process, versus a child of the load balancer process, otherwise they're sharing the same JVM instance I think, which has clustering implications.

Yes, they should be separate processes.

> (heck, I'd like to support two load balancing instances, backed by 2 or 3 app node instances) that should be pretty fault tolerant, besides good for testing clustering stuff.

Yes, exactly. That is the idea.  A good concept to follow is a mesh router.  Where you can have many load balancer that can balance the mesh of nodes.  Any hit to the balancer goes into the mesh.

An important fact here is the ability to scale up/down the nodes in a particular balancer.  This way, we can create commands to create 3 more nodes and attach them to the balancer.  Is this possible?

**Comment by bdw429s** [comment]
Hey @denny looking to play with this today.  Any more thoughts on querying, adding, and removing nodes from the cluster once it's running? I think we may need a management port on the LB to connect to.

**Comment by bdw429s** [comment]
@Denny, I tried the method of load balancing existing servers like so 
```
--loadbalance 127.0.0.1:54965,127.0.0.1:55313,www.google.com:80
```
and Runwar threw this error:
```
The server for C:\sandbox\cluster\ is starting on 127.0.0.1:55731...
USAGE:   java -jar runwar.jar [-war] path/to/war [options]
Must specify -war path/to/war, or -stop [-stop-socket]
```

**Comment by bdw429s** [comment]
@denny, I added the `-war` option back in and received this error:

```
The server for C:\sandbox\cluster\ is starting on 127.0.0.1:55731...           Apr 11, 2017 4:18:56 PM runwar.logging.Logger log                              INFO: Starting instances...                                                    Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: 2                 at runwar.Start.launchServer(Start.java:33)                                    at runwar.Start.main(Start.java:49)                                    
```

Please let me know how to use this.

**Comment by denuno** [comment]
It required a war before, I just pushed a build that fixes that, and gives a better error message if the balance host isn't correct.  Try it like this:
`--loadbalance "http://127.0.0.1:54965,http://127.0.0.1:55313,http://www.google.com:80"`
Or just:
`--loadbalance "http://www.yahoo.com:80,http://www.google.com:80"`
and then hit the load balancer with curl, and you should see the content flip between the sites (there's some sticky session handling built in I think that might trip ya up in a browser).

Also it'll now use the passed in port for the balancer.

The balancer "admin" is hard-coded to port 9080, so http://127.0.0.1:9080/ will show you the super-cheese-but-works-to-add-remove-hosts admin.

**Comment by bdw429s** [comment]
Thanks @denuno, I got the balancer to work and it appeared to round robin in a browser if I started the servers myself and then started the balancer.  I wasn't able to get it to work when I tried to have it start the servers and the balancer.  One server icon came up in the system tray and the logs showed two servers starting, but I couldn't get any ports at all to load the browser.  I'm not sure if I'm using the right combination of args for Runwar to work.
